core/src/main/java/com/linecorp/armeria/client/retry/RetryConfigBuilder.java

```diff
@@ -0,0 +1,144 @@
+        this.retryRule = checkNotNull(retryRule);
+        retryRuleWithContent = null;
+        maxContentLength = 0;
+    }
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRuleWithContent}.
+     */
+    public RetryConfigBuilder(RetryRuleWithContent<T> retryRuleWithContent) {
+        this(retryRuleWithContent, Integer.MAX_VALUE);
+    }
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRuleWithContent} and maxContentLength.
+     */
+    public RetryConfigBuilder(RetryRuleWithContent<T> retryRuleWithContent, int maxContentLength) {
+        retryRule = null;
+        this.retryRuleWithContent = checkNotNull(retryRuleWithContent);
+        checkArgument(maxContentLength > 0,
+                      "maxContentLength: %s (expected: > 0)", maxContentLength);
+        this.maxContentLength = maxContentLength;
+    }
+
+    /**
+     * Sets maxTotalAttempts.
+     */
+    public RetryConfigBuilder<T> maxTotalAttempts(int maxTotalAttempts) {
+        checkArgument(
+                maxTotalAttempts > 0,
+                "maxTotalAttempts: %s (expected: > 0)",
+                maxTotalAttempts);
+        this.maxTotalAttempts = maxTotalAttempts;
+        return this;
+    }
+
+    /**
+     * Sets responseTimeoutMillisForEachAttempt.
+     */
+    public RetryConfigBuilder<T> responseTimeoutMillisForEachAttempt(long responseTimeoutMillisForEachAttempt) {
+        checkArgument(
+                responseTimeoutMillisForEachAttempt >= 0,
+                "responseTimeoutMillisForEachAttempt: %s (expected: >= 0)",
+                responseTimeoutMillisForEachAttempt);
+        this.responseTimeoutMillisForEachAttempt = responseTimeoutMillisForEachAttempt;
+        return this;
+    }
+
+    /**
+     * Sets responseTimeoutMillisForEachAttempt by converting responseTimeoutForEachAttempt to millis.
+     */
+    public RetryConfigBuilder<T> responseTimeoutForEachAttempt(Duration responseTimeoutMillisForEachAttempt) {
+        final long millis = responseTimeoutMillisForEachAttempt.toMillis();
+        checkArgument(
+                millis >= 0,
+                "responseTimeoutForEachAttempt.toMillis(): %s (expected: >= 0)",
+                millis);
+        this.responseTimeoutMillisForEachAttempt = millis;
+        return this;
+    }
+
+    /**
+     * Builds a {@link RetryConfig} from this builder's values and returns it.
+     */
+    public RetryConfig<T> build() {
+        if (retryRule != null) {
+            return new RetryConfig<>(retryRule, maxTotalAttempts, responseTimeoutMillisForEachAttempt);
+        } else if (retryRuleWithContent != null) {
+            return new RetryConfig<>(
+                    retryRuleWithContent,
+                    maxContentLength,
+                    maxTotalAttempts,
+                    responseTimeoutMillisForEachAttempt);
+        } else {
+            throw new IllegalStateException("RetryConfigBuilder must have a rule.");
```

Comment:
Remove this `else` block? I think we can reach this condition.

Reply:
Done